File: 1103_distribute_candies/distribute.go
package main

// Time: O(n)
// Space: O(n)
func distributeCandies(candies int, num_people int) []int {
	res := make([]int, num_people)
	index, count := 0, 1
	for candies > 0 {
		if count < candies {
			res[index%num_people] += count
		} else {
			res[index%num_people] += candies
		}
		candies -= count
		count++
		index++
	}
	return res
}

func main() {

}

File: 1103_distribute_candies/distribute.java
// Time: O(n)
// Space: O(n)
class Solution {
    public int[] distributeCandies(int candies, int num_people) {
        int[] res = new int[num_people];
        int count = 1, i = 0;
        while (candies > 0) {
            res[i++] += (count <= candies) ? count : candies;
            candies -= count++;
            if (i >= num_people)
                i = 0;
        }
        return res;
    }
}